components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/handlers/AbstractResponseTypeHandler.java

```diff
@@ -101,6 +103,19 @@ public boolean validateScope(OAuthAuthzReqMessageContext oauthAuthzMsgCtx) throw
                 .getAuthorizationCodeValidityPeriod());
         oauthAuthzMsgCtx.setAccessTokenIssuedTime(scopeValidationCallback.getAccessTokenValidityPeriod());
         oauthAuthzMsgCtx.setApprovedScope(scopeValidationCallback.getApprovedScope());
+        // Deriving the global level scope validator implementations.
```

Comment:
Need to add the logic to handle allowed scopes.

Reply:
Added in upper level (AuthorizationHandlerManager) in https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1468/commits/328d264a2e70a4028ddfcbf05a25f57e3e873174